egs/wsj/s5/steps/nnet3/nnet3_train_lib.py

```diff
@@ -0,0 +1,622 @@
+
+def ComputeAveragePosterior(dir, iter, egs_dir, num_archives,
+                            prior_subset_size, run_opts):
+    # Note: this just uses CPUs, using a smallish subset of data.
+    """ Computes the average posterior of the network"""
+    import glob
+    for file in glob.glob('{0}/post.{1}.*.vec'.format(dir, iter)):
+        os.remove(file)
+    
+    if run_opts.num_jobs_compute_prior > num_archives:
+        egs_part = 1
+    else:
+        egs_part = 'JOB'
+
+    RunKaldiCommand("""
+{command} JOB=1:{num_jobs_compute_prior} {prior_queue_opt} {dir}/log/get_post.{iter}.JOB.log \
+    nnet3-subset-egs --srand=JOB --n={prior_subset_size} ark:{egs_dir}/egs.{egs_part}.ark ark:- \| \
+    nnet3-merge-egs --measure-output-frames=true --minibatch-size=128 ark:- ark:- \| \
+    nnet3-compute-from-egs {prior_gpu_opt} --apply-exp=true \
+  "nnet3-am-copy --raw=true {dir}/combined.mdl -|" ark:- ark:- \| \
+matrix-sum-rows ark:- ark:- \| vector-sum ark:- {dir}/post.{iter}.JOB.vec
+    """.format(command = run_opts.command,
+               dir = dir,
+               num_jobs_compute_prior = run_opts.num_jobs_compute_prior,
+               prior_queue_opt = run_opts.prior_queue_opt,
+               iter = iter, prior_subset_size = prior_subset_size,
+               egs_dir = egs_dir, egs_part = egs_part,
+               prior_gpu_opt = run_opts.prior_gpu_opt))
+
+    # make sure there is time for $dir/post.{iter}.*.vec to appear.
+    time.sleep(5)
+    avg_post_vec_file = "{dir}/post.{iter}.vec".format(dir=dir, iter=iter)
+    RunKaldiCommand("""
+{command} {dir}/log/vector_sum.{iter}.log \
+    vector-sum {dir}/post.{iter}.*.vec {output_file}
+        """.format(command = run_opts.command,
+                   dir = dir, iter = iter, output_file = avg_post_vec_file))
+
+    for file in glob.glob('{0}/post.{1}.*.vec'.format(dir, iter)):
+        os.remove(file)
+    return avg_post_vec_file
+
+def AdjustAmPriors(dir, input_model, avg_posterior_vector, output_model, run_opts):
+    RunKaldiCommand("""
+{command} {dir}/log/adjust_priors.final.log \
+nnet3-am-adjust-priors {input_model} {avg_posterior_vector} {output_model}
+    """.format(command = run_opts.command,
+               dir = dir, input_model = input_model,
+               avg_posterior_vector = avg_posterior_vector,
+               output_model = output_model))
+
+def RemoveEgs(egs_dir):
+    RunKaldiCommand("steps/nnet2/remove_egs.sh {egs_dir}".format(egs_dir))
+
+def CleanNnetDir(nnet_dir, num_iters, egs_dir, num_iters_combine = None,
+                 preserve_model_interval = 100,
+                 remove_egs = True):
+    try:
+        if remove_egs:
+            RemoveEgs(egs_dir)
+
+        for iter in range(num_iters):
+            print("deleting iter {0}".format(iter))
+            RemoveModel(nnet_dir, iter, num_iters, 1,
+                        preserve_model_interval)
+    except (IOError, OSError) as err:
```

Comment:
I am catching the exception just to add some logging info, nothing more than that.